Lib/ast.py

```diff
@@ -1053,15 +1054,77 @@ def visit_AsyncWith(self, node):
         with self.block(extra=self.get_type_comment(node)):
             self.traverse(node.body)
 
+    def _str_literal_helper(
+        self, value, quote_types=("'", '"', '"""', "'''"), escape=""
+    ):
+        """Helper for writing string literals, minimising escapes.
+        Returns (possible quote types, string literal to write).
+        """
+        # Escape characters we've been told to escape, backslashes, and
+        # non-printable characters other than \n and \t.
+        escape = {*escape, '\\'}
+        val = "".join(
+            c.encode('unicode_escape').decode('ascii')
+            if c in escape or (not c.isprintable() and c not in '\n\t') else c
+            for c in value
+        )
+        qts = [quote for quote in quote_types if quote not in val]
+        if "\n" in val:
+            qts = [quote for quote in qts if quote in ('"""', "'''")]
```

Comment:
Please use descriptive names, such as `possible_quotes`
```suggestion
        possible_quotes = [quote for quote in quote_types if quote not in escaped_string]
        if "\n" in escaped_string:
            # For multi-line strings, use triple quotes
            possible_quotes = [
                quote for quote in possible_quotes if len(quote) == 3
            ]
```